app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/event_parser.rb

```diff
@@ -2,26 +2,25 @@ module ManageIQ::Providers::IbmCloud::VPC::CloudManager::EventParser
   def self.event_to_hash(event, ems_id)
     event_hash = {
       :event_type => event["action"],
-      :source     => "IBM_CLOUD_VPC",
+      :source     => "IBMCloud-VPC",
       :ems_id     => ems_id,
       :ems_ref    => event["_id"],
       :timestamp  => event["eventTime"],
       :full_data  => event
     }
 
     case event_hash[:event_type]
-    when 'is.instance.instance.create'
+    when /^is\.instance/
       parse_vm_event!(event, event_hash)
     end
 
     event_hash
   end
 
   def self.parse_vm_event!(event, event_hash)
-    return if event['responseData'].nil?
+    return if event.dig('target', 'id').partition('instance:').last == ""
 
-    event_hash[:vm_uid_ems] = event.dig('responseData', 'id')
-    event_hash[:vm_ems_ref] = event.dig('responseData', 'id')
-    event_hash[:vm_name]    = event.dig('responseData', 'name')
+    event_hash[:vm_uid_ems] = event.dig('target', 'id').partition('instance:').last
+    event_hash[:vm_ems_ref] = event.dig('target', 'id').partition('instance:').last
```

Comment:
This had to be changed because some instance events did not contain the vm ref and those that did contain the ref did not have it readily available in `responseData` as was the case for `is.instance.instance.create`. Therefore the ref is now obtained by parsing it from a longer target ID.

i.e. 
```
crn:v1:bluemix:public:is:us-south-2:a/c56c9a268d23e1b339ac14774358133c::instance:0727_3fa23c25-3e03-46f3-a503-d459953816ef
```

If there is a better way of going about this let me know

Reply:
I think a regular expression would be safer here:
```
.match(/::instance:(?<instance_id>\S+)$/).named_captures["instance_id"]
```

Reply:
You might want to pull this out into a method like
```
instance_id = parse_event_instance_id(event)
return if instance_id.nil?
```